air/src/proof/ood_frame.rs

```diff
@@ -149,13 +194,68 @@ impl Deserializable for OodFrame {
         let num_trace_state_bytes = source.read_u16()? as usize;
         let trace_states = source.read_vec(num_trace_state_bytes)?;
 
+        // read Lagrange kernel column trace rows
+        let num_lagrange_state_bytes = source.read_u16()? as usize;
+        let lagrange_kernel_trace_states = source.read_vec(num_lagrange_state_bytes)?;
+
         // read constraint evaluations row
         let num_constraint_evaluation_bytes = source.read_u16()? as usize;
         let evaluations = source.read_vec(num_constraint_evaluation_bytes)?;
 
         Ok(OodFrame {
             trace_states,
+            lagrange_kernel_trace_states,
             evaluations,
         })
     }
 }
+
+// OOD FRAME TRACE STATES
+// ================================================================================================
+
+/// Stores the trace evaluations at `z` and `gz`, where `z` is a random Field element. If
+/// the Air contains a Lagrange kernel auxiliary column, then that column interpolated polynomial
+/// will be evaluated at `z`, `gz`, `g^2 z`, ... `g^(2^(v-1)) z`, where `v == log(trace_len)`, and
+/// stored in `lagrange_kernel_frame`.
+pub struct OodFrameTraceStates<E: FieldElement> {
```

Comment:
Is this struct still used somewhere?

Reply:
It is used [here](https://github.com/plafer/winterfell/blob/a3377d936b32b196e4a59a87ad39dc27cedffcb8/prover/src/trace/poly_table.rs#L80), [here](https://github.com/plafer/winterfell/blob/a3377d936b32b196e4a59a87ad39dc27cedffcb8/air/src/proof/ood_frame.rs#L57) and [here](https://github.com/plafer/winterfell/blob/a3377d936b32b196e4a59a87ad39dc27cedffcb8/prover/src/composer/mod.rs#L66)